app/controllers/v0/virtual_agent/virtual_agent_appeal_controller.rb

```diff
@@ -37,6 +37,10 @@ def get_appeal_from_lighthouse(ssan, user_name)
         http.use_ssl = true
 
         response = http.request(req)
+        if response.code != '200'
+          Rails.logger.error "Lighthouse API returned #{response.code}: #{response.body}"
+          raise "#{response.code}: #{response.body}"
```

Comment:
if the requirements are to raise this error only in staging, why wouldn't we simply add a `if Rails.env.staging?` on the end of the line?

Reply:
The highlighted part of code in VirtualAgentAppealController class, gets executed only when the function get_appeal_from_lighthouse is called and this function is called in line# 12 only if the environment is 'staging' based on line# 9 (as that is the only environment where Lighthouse's sandbox env is called to fetch appeals data for test users). 
In all other environments line# 16 and 17 are used, which do not use Rails.logger.error.

Reply:
I understand your point that the code path prevents this exception from being raised for any reason other than the env being staging. 

In that case, I think the method `#get_appeal_from_lighthouse` should be renamed so it's explicit that it's a method for staging env only.

Reply:
I also think we need to introduce at least a unit test for this new functionality. A unit test shows the next person who reads your code what you intended. I think my experience is emblematic of the confusion that is possible.